test/infrastructure/vcsim/README.md

```diff
@@ -1,4 +1,156 @@
+of the cluster.
+
+The picture below explains how this works in detail:
+
+![Architecture](architecture-part2.drawio.svg)
+
+- When the cluster API controllers (KCP, MD controllers) creates a `VSphereMachine`, the CAPV controllers first create
+  a `VSphereVM` resource. 
+- The, when reconciling the `VSphereVM` CAPV controllers, use the govmomi library to connect to vCenter (in this case to vcsim)
+  to provision a VM.
+
+Given that VM provisioned by vcsim are fake, there won't be cloud-init running on the machine, and thus it is responsibility 
+of the vmBootstrapReconciler component inside the vcsim controller to "mimic" the machine bootstrap process:
+NOTE: This process is implemented using the code for CAPI's in memory provider.
+
+- If the machine is a control plane machine, fake static pods are created for API server, etcd and other control plane
+  components (controller manager and scheduler are omitted from the picture for sake of simplicity).
+- API server fake pods are registered as "backend" for the cluster VIP, as well as etcd fake pods are registered as
+  "members" of a fake etcd cluster.
+
+Cluster VIP, fake API server pods and fake Etcd pods provide a minimal fake Kubernetes control plane, just capable of the
+operations required to trick Cluster API in believing a real K8s cluster is there.
+
+Finally, in order to complete the provisioning of the fake VM, it is necessary to assign an IP to it. This task is
+performed by the vmIPReconciler component inside the vcsim controller.
+
+## Working with vcsim
+
+### Tilt
+
+vcsim can be used with Tilt for local development.
+
+To use vcsim it is required to add it the list of enabled providers in your `tilt-setting.yaml/json`; you can also
+provide extra args or enable debugging for this provider e.g.
+
+```yaml
+...
+provider_repos:
+  - ../cluster-api-provider-vsphere
+enable_providers:
+  - kubeadm-bootstrap
+  - kubeadm-control-plane
+  - vsphere
+  - vcsim
+extra_args:
+  vcsim:
+    - "--v=2"
+    - "--logging-format=json"
+debug:
+  vcsim:
+    continue: true
+    port: 30040
+...
+```
+
+Note: vcsim is not a Cluster API provider, however, for sake of convenience we are "disguising" it as a Cluster API
+runtime extension, so it is possible to deploy it leveraging on the existing machinery in Tilt (as well as in E2E tests).
+
+After starting tilt with `make tilt-up`, you can use the [vcsim.sh](scripts/vcsim.sh) script and the instruction above
+to create a test cluster.
+
+See [Developing Cluster API with Tilt](https://cluster-api.sigs.k8s.io/developer/tilt) for more details.
+
+### E2E tests
+
+vsim can be used to run a subset of CAPV E2E tests that can be executed by setting `GINKGO_FOCUS="\[vcsim\]"`.
+
+See [Running the end-to-end tests locally](https://cluster-api.sigs.k8s.io/developer/testing#running-the-end-to-end-tests-locally) for more details.
+
+Note: The code for the E2E test setup will take care of creating the `VCenterSimulator`, the `ControlPlaneEndpoint`
```

Comment:
Just to confirm.

If I want to run the e2e tests against tilt, I just need the tilt settings from above and the GINKGO_FOCUS and everything else is done automatically?

Reply:
(And same for supervisor mode, just with different tilt-settings & GINKGO_FOCUS?)

Reply:
Yes, it is correct.
When running E2E in tilt mode it is up to you to configure Tilt with the necessary providers.
The E2E, recognizes a management cluster is there, then it skips creating the kind cluster installing the providers, but everything else is the same.

Reply:
Thx!